docs/architecture.md

```diff
@@ -0,0 +1,60 @@
+# Architecture
```

Comment:
IMO this file can get merged with API-and-internals.md

and since readme and contributing are in root, i kinda prefer keeping this md file there rather than docs/. wdyt?